hs_core/hydroshare/utils.py

```diff
@@ -239,6 +239,82 @@ def get_fed_zone_files(irods_fnames):
     return ret_file_list
 
 
+def get_file_from_irods(res_file):
+    """
+    Copy the file (res_file) from iRODS (local or federated zone)
+    over to django (temp directory) which is
+    necessary for manipulating the file (e.g. metadata extraction).
+    Note: The caller is responsible for cleaning the temp directory
+
+    :param res_file: an instance of ResourceFile
+    :return: location of the copied file
+    """
+    res = res_file.resource
+    if res_file.resource_file or res_file.fed_resource_file:
+        istorage = IrodsStorage()
```

Comment:
@pkdash when ```res_file.fed_resource_file``` is not empty, ```istorage = IrodsStorage('federated')``` needs to be called. I will write code snippet for this function for your reference.